notebooks/00-Walkthrough-minicircle.ipynb

```diff
@@ -0,0 +1,894 @@
+    "dictionaries containing the grain data to be plotted. We therefore instantiate (\"set-up\") the `grain_stats` dictionary\n",
+    "to hold these results and unpack each to the keys `statistics` and `plots` respectively.\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b1de4c2c-b6df-4c0a-b829-d70111d31693",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "grainstats = GrainStats(\n",
+    "    data=filtered_image.images[\"gaussian_filtered\"],\n",
+    "    labelled_data=grains.directions[\"upper\"][\"labelled_regions_02\"],\n",
+    "    pixel_to_nanometre_scaling=filtered_image.pixel_to_nm_scaling,\n",
+    "    direction=\"upper\",\n",
+    "    base_output_dir=\"grains\",\n",
+    "    image_name=filtered_image.filename.stem,\n",
+    "    **grainstats_config,\n",
+    ")\n",
+    "_temp = grainstats.calculate_stats()\n",
+    "grain_stats = {\"statistics\": _temp[0], \"plots\": _temp[1]}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "893a3891-dbb9-45e5-9c7a-a3ee7160744d",
+   "metadata": {},
+   "source": [
+    "The `statistics` is a [Pandas\n",
+    "DataFrame](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.html). We can print this out as\n",
+    "shown below."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "21a003a3-49ca-401c-9173-2c60ea56f329",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "grain_stats[\"statistics\"]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f2ebf89b-4e1e-4a8c-b1b7-bb9343f2c6ea",
+   "metadata": {},
+   "source": [
+    "Further we can summarise the dataframe or a subset of varaibles."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ee090ac5-6023-4fdd-835a-e58ee184316a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "grain_stats[\"statistics\"][[\"smallest_bounding_width\", \"aspect_ratio\"]].describe()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c2ce88ce-9382-45a4-af4e-924aa28d9540",
+   "metadata": {},
+   "source": [
+    "### Plotting Individual Grains\n",
+    "\n",
+    "It is possible to plot the individual grains in the same way that whole images are plotted. These are now stored as the\n",
+    "`grain_stats[\"plots\"]` dictionary. We can find out how many grains there are either by looking at the number of rows\n",
+    "reported when the statisics Pandas Data Frame was printed above or its summary (from `.describe()`), which reports the\n",
+    "`count`.\n",
+    "\n",
+    "We can then plot any of these using the number indexing of the list. We using\n",
+    "[f-strings](https://realpython.com/python-f-strings/) as the value for the argument `filename` to fill in the images\n",
+    "filename automatically and append the `_grain#` for the image numbe we are plotting, in this example `0`. Try plotting\n",
+    "some other images."
```

Comment:
I noticed this problem this morning when starting to investigate the line issues and think the issue lies deeper and so rather than patching things here we need to change the structure. I've written this up in #491.